PcapDotNet/src/PcapDotNet.Packets/Dhcp/Options/DhcpOption.cs

```diff
@@ -0,0 +1,363 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using PcapDotNet.Base;
+
+namespace PcapDotNet.Packets.Dhcp.Options
+{
+    /// <summary>
+    /// RFC 2132.
+    /// Abstract class for all possible Dhcp-Options
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
+    public abstract class DhcpOption : IEquatable<DhcpOption>
+    {
+        /// <summary>
+        /// Option-Code according RFC 2132
+        /// </summary>
+        public DhcpOptionCode OptionCode
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Length of the Dhcp-Option
+        /// </summary>
+        public abstract byte Length
+        {
+            get;
+        }
+
+        /// <summary>
+        /// create new Option
+        /// </summary>
+        /// <param name="code">Option-Code</param>
+        protected DhcpOption(DhcpOptionCode code)
+        {
+            OptionCode = code;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1502:AvoidExcessiveComplexity")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
```

Comment:
Consider solving these maintainability issues by using registration mechanism for the options, similar to what is being done in other protocols.